modules/geoip-lite-2024-07-19/lib/geoip.js

```
@@ -0,0 +1,540 @@
+		if (floor <= ip && ceil >= ip) {
+			geodata.range = [floor, ceil];
+
+			if (recordSize === RECORD_SIZE) {
+				geodata.country = buffer.toString('utf8', (line * recordSize) + 8, (line * recordSize) + 10);
+			} else {
+				locId = buffer.readUInt32BE((line * recordSize) + 8);
+
+				// -1>>>0 is a marker for "No Location Info"
+				if(-1>>>0 > locId) {
+					geodata.country = locBuffer.toString('utf8', (locId * locRecordSize) + 0, (locId * locRecordSize) + 2).replace(/\u0000.*/, '');
+					geodata.region = locBuffer.toString('utf8', (locId * locRecordSize) + 2, (locId * locRecordSize) + 5).replace(/\u0000.*/, '');
```

Biome

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)